### ISSUE TITLE
Open source local provider

### ISSUE THREAD

**Issue by FerronN** [Piwigo/piwigo-tag-recognition#6]
Hi @Zacharieg,

Thank you for writing this extension! I'm going to run Piwigo and face tag is a must have.
I have Azure and could use that but I'm always for the most secure option and that's running is peace of software myself.

I'm looking to use [deepface](https://github.com/serengil/deepface) or [compreface](https://github.com/exadel-inc/CompreFace). Both supply a simple REST api. I'm going to run one of these two in a docker container like piwigo itself.

What do you think? Could we work together to implement one of these products and there api for a self-hosted solution?

**Comment by Zacharieg** [comment]
Hi FerronN !

You're welcome ! Unfortunately, I don't work in the project anymore... But I developped the app thinking of different ways to generate the tags !

I encourage you to clone the repository and try to adapt it, create a new _api_class_ in [this](https://github.com/Piwigo/piwigo-tag-recognition/tree/main/api_classes) folder.

If you have any question, you can ask them here, and if you succeed to developp it, don't hesitate to do a merge request to let the other enjoy your work !

**Comment by Zacharieg** [comment]
So you have to : 
**Create a new file in the _api_classes_ folder (you can copy an existing file)**
The functions you have to write here are : **getInfo** to add information about your API (here it should be like "self hosted api), **getConfParams** to ask the user some information like the url of your selfhosted API, and finally **generateTags** a function that should return an array with all the tags

**Inform the plugin of your new API**
In [this](https://github.com/Piwigo/piwigo-tag-recognition/blob/main/include/api_types.php) file, add your API name (should be the same file and class name) in the TR_API_LIST

After that it should work, notice me if you find bugs

**Comment by FerronN** [comment]
Thank you for the description! I'm going to try to make it work :)

**Comment by FerronN** [comment]
The extension is not specifically designed for face recognition right (if I look at the code)?

**Comment by Zacharieg** [comment]
Hum at the start no you're right, but in fact the input is an image and the output is an array of text so you can do what ever you want.

**Comment by FerronN** [comment]
True, but then I would need a page in Piwigio to show all recognized faces so you can name them. Any suggestions how to approach this?

**Comment by Zacharieg** [comment]
With my extension yeah it should be hard to do, maybe with [this](https://piwigo.org/ext/extension_view.php?eid=910) plugin ?

**Comment by 3JL** [comment]
@FerronN, could you test this code? Let me know if you need help! [https://github.com/Piwigo/piwigo-tag-recognition/issues/7](url)
